core/controllers/acl_decorators.py

```diff
@@ -303,6 +303,68 @@ def test_can_edit(self, collection_id, **kwargs):
     return test_can_edit
 
 
+def can_edit_suggestion(handler):
+    """Decorator to check whether the user can edit given suggestion.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that now also checks if
+            a user has permission to edit a given suggestion.
+    """
+
+    def test_can_edit_suggestion(self, **kwargs):
+        """Checks if the user can edit the suggestion.
+
+        Args:
+            **kwargs: dict(str, str). Credentials to check for permisssion.
+
+        Returns:
+            *. The return value of the decorated function.
+
+        Raises:
+            NotLoggedInException: The user is not logged in.
+            PageNotFoundException: The page is not found.
+            UnauthorizedUserException: The user does not have
+                credentials to edit a suggestion.
+        """
+        credentials_to_check = kwargs
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+
+        users_which_having_rights = (
+            suggestion_services.check_can_edit_suggestion(
+                credentials_to_check, self.user_id))
+
+        if users_which_having_rights['author']:
+            return handler(self, credentials_to_check['suggestion_id'])
+
+        elif credentials_to_check['target_type'] == 'exploration':
+            if users_which_having_rights['page_not_found']:
+                raise base.UserFacingExceptions.PageNotFoundException
+
+            if users_which_having_rights['exploration_author']:
+                return handler(self, credentials_to_check['suggestion_id'])
+            else:
+                raise base.UserFacingExceptions.UnauthorizedUserException(
+                    'You do not have credentials to edit this suggestion.')
+
+        elif credentials_to_check['target_type'] == 'collection':
```

Comment:
You can get the target_type directly from the suggestion (get the suggestion object, and check the target_type).

Reply:
done, and sorry for this